lib/tasks/generateResourcesJson.js

```diff
@@ -0,0 +1,49 @@
+"use strict";
+
+// const log = require("@ui5/logger").getLogger("builder:tasks:generateResourcesJson");
+const resourceListCreator = require("../processors/resourceListCreator");
+
+const DEFAULT_EXCLUDES = [
+	"!**/.DS_Store",
+	"!**/.eslintrc",
+	"!**/.eslintignore",
+	"!**/.gitignore"
+];
+
+/**
+ * Task for creating a library resources.json, describing all productive resources in the library.
+ *
+ * @public
+ * @alias module:@ui5/builder.tasks.generateResourcesJson
+ * @param {Object} parameters Parameters
+ * @param {module:@ui5/fs.DuplexCollection} parameters.workspace DuplexCollection to read and write files
+ * @param {module:@ui5/fs.AbstractReader} parameters.dependencies Reader or Collection to read dependency files
+ * @param {Object} parameters.options Options
+ * @param {string} parameters.options.projectName Project name
+ * @returns {Promise<undefined>} Promise resolving with <code>undefined</code> once data has been written
+ */
+module.exports = async function({workspace, dependencies, options}) {
+	let resources;
+	if (workspace.byGlobSource) { // API only available on duplex collections
+		resources = await workspace.byGlobSource(["/resources/**/*.*", ...DEFAULT_EXCLUDES]);
+		// HACK add resources from internal writer of workspace
+		const writtenResources = await workspace._writer.byGlob(["/resources/**/*.*", ...DEFAULT_EXCLUDES]);
+		writtenResources.forEach((res) => {
+			if ( resources.indexOf(res) < 0 ) {
+				resources.push(res);
+			}
+		});
+	} else {
```

Comment:
`workspace.byGlob` also only returns resources of the current project. Searching with the namespace should be possible by now (I know your comment is from last year), since with v2.0 all projects are required to have a namespace.

Reply:
And what about `sap.ui.core`? Yet another, hard-coded exception?

Reply:
Yes, now I get it. Indeed for `sap.ui.core` the namespace `sap/ui/core` is insufficient. A glob pattern like to current one should be fine though.

Still, why do we use `byGlobSource` instead of `byGlob`? I thought we are also interested in generated files.

Reply:
byGlobSource retrieves the debug files as well, byGlob does not retrieve them

Reply:
> byGlobSource retrieves the debug files as well.

Are you sure? I think that should not be the case unless they already exist in the source directory.

Reply:
you're right, byGlob retrieves all the files.

adjusted logic for copy from source such that it does not depend on the combination of byGlobSource and adding missing ones from byGlob on top.
https://github.com/SAP/ui5-builder/pull/390/commits/f346f3e2e827e97fab0f32160a85bb84c8409251